libs/coin-framework/src/bridge/jsHelpers.ts

```diff
@@ -545,8 +546,21 @@ export function makeAccountBridgeReceive<A extends Account = Account>(
       getAddressFn(deviceId, arg).then(r => {
         const accountAddress = account.freshAddress;
 
-        if (verify && r.address !== accountAddress) {
-          throw new WrongDeviceForAccount();
+        if (verify) {
+          // NOTE: could pass a param to make different checks for different currencies
+          if (account.currency.id === "hedera") {
```

Comment:
We do not want this kind of check in common code, is there another way to do this ?

Reply:
Good point, maybe by passing additional params to make the verify happen on the provided vars instead, wdyt?. 
For now we could start with a boolean `verifyUsingPublicKey`, as hedera seems to be the sole chain to do a different so. 

So something like 
```js
if (verify) {
          // NOTE: could pass a param to make different checks for different currencies
          if (verifyUsingPublicKey && r.publicKey !== account.seedIdentifier) {
```